src/services/table/tableUtils.tsx

```diff
@@ -0,0 +1,126 @@
+import { SearchOutlined } from '@ant-design/icons';
+import { Input } from 'antd';
+import {
+  ColumnFilterItem,
+  ColumnTitle,
+  FilterConfirmProps,
+  FilterDropdownProps,
+  Key,
+} from 'antd/lib/table/interface';
+import { DataIndex } from 'rc-table/lib/interface';
+import React, { MutableRefObject, ReactNode } from 'react';
+
+import { InterviewQuestionModel } from '../Client';
+import { getDistinctValues, toPascalCase } from '../stringUtils';
+import FilterDropdown from './FilterDropdown';
+
+export function createDefaultColumnProps(dataIndex: string): {
+  title: string;
+  dataIndex: string;
+  key: string;
+} {
+  return {
+    title: toPascalCase(dataIndex),
+    dataIndex,
+    key: dataIndex,
+  };
+}
+
+const searchHighlightColor = '#1890ff';
+
+export function createColumnSearchProps(
+  dataIndex: 'title' | 'category' | 'content' | 'difficulty',
+  searchInput: MutableRefObject<Input | undefined>,
+  handleSearchCallback: (
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ) => void,
+  handleResetCallback: (
+    clearFilters: () => void,
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ) => void
+): {
+  title: ColumnTitle<InterviewQuestionModel>;
```

Comment:
I recommend move definition of props into separate file. It's hard to read code now.